muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/WorkspaceForumRESTService.java

```diff
@@ -474,11 +477,11 @@ public Response createThread(@PathParam ("WORKSPACEENTITYID") Long workspaceEnti
     }
 
     if (sessionController.hasWorkspacePermission(ForumResourcePermissionCollection.FORUM_WRITE_WORKSPACE_MESSAGES, workspaceEntity)) {
-      if (Boolean.TRUE.equals(newThread.getSticky()) || Boolean.TRUE.equals(newThread.getLocked())) {
+      if (Boolean.TRUE.equals(newThread.getSticky()) || LockForumThread.valueOf(newThread.getLock()) == LockForumThread.ALL) {
```

Comment:
Is it guaranteed that `newThread.getLock()` is not `null`? `valueOf` won't deal with null values.